com.unity.netcode.gameobjects/Runtime/Serialization/FastBufferWriter.cs

```diff
@@ -0,0 +1,1042 @@
+                    case TypeCode.Byte:
+                        WriteValueSafe((byte)value);
+                        break;
+                    case TypeCode.Int16:
+                        WriteValueSafe((short)value);
+                        break;
+                    case TypeCode.UInt16:
+                        WriteValueSafe((ushort)value);
+                        break;
+                    case TypeCode.Int32:
+                        WriteValueSafe((int)value);
+                        break;
+                    case TypeCode.UInt32:
+                        WriteValueSafe((uint)value);
+                        break;
+                    case TypeCode.Int64:
+                        WriteValueSafe((long)value);
+                        break;
+                    case TypeCode.UInt64:
+                        WriteValueSafe((ulong)value);
+                        break;
+                }
+                return;
+            }
+            if (value is GameObject)
+            {
+                WriteValueSafe((GameObject)value);
+                return;
+            }
+            if (value is NetworkObject)
+            {
+                WriteValueSafe((NetworkObject)value);
+                return;
+            }
+            if (value is NetworkBehaviour)
+            {
+                WriteValueSafe((NetworkBehaviour)value);
+                return;
+            }
+            if (value is INetworkSerializable)
+            {
+                //TODO ((INetworkSerializable)value).NetworkSerialize(new NetworkSerializer(this));
+                return;
+            }
+
+            throw new ArgumentException($"{nameof(FastBufferWriter)} cannot write type {value.GetType().Name} - it does not implement {nameof(INetworkSerializable)}");
+        }
+
+        /// <summary>
+        /// Write an INetworkSerializable
+        /// </summary>
+        /// <param name="value">The value to write</param>
+        /// <typeparam name="T"></typeparam>
+        public void WriteNetworkSerializable<T>(in T value) where T : INetworkSerializable
+        {
+            // TODO
+        }
+
+        /// <summary>
+        /// Get the required amount of space to write a GameObject
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public static int GetWriteSize(GameObject value)
+        {
+            return sizeof(ulong);
+        }
+
+        /// <summary>
+        /// Get the required amount of space to write a GameObject
+        /// </summary>
+        /// <returns></returns>
+        public static int GetGameObjectWriteSize()
+        {
+            return sizeof(ulong);
+        }
+
+        /// <summary>
+        /// Write a GameObject
+        /// </summary>
+        /// <param name="value">The value to write</param>
+        public void WriteValue(GameObject value)
+        {
+            var networkObject = (value).GetComponent<NetworkObject>();
```

Comment:
Hey @ShadauxCat I suggest converting this GetComponent call to TryGetComponent, as the latter does not allocate anything (in Editor)